src/components/auth/login/Login.jsx

```diff
@@ -1,38 +1,49 @@
 import React, { useState } from "react";
 import { auth } from "../../config/firebaseConfig";
 import db from "../../config/firebaseConfig";
-import AddProducts from "../../addProductsForm/AddProducts";
 import Button from "react-bootstrap/Button";
-import { Col, Row } from "react-bootstrap";
+import { Col } from "react-bootstrap";
 import Form from "react-bootstrap/Form";
+import {
+  logIn,
+  logOut,
+  logInError,
+  popUpStatus,
+} from "../../redux/actions/index";
+import { useSelector, useDispatch } from "react-redux";
 
 const LogInForm = () => {
-  const [isAdmin, setIsAdmin] = useState(false);
-  const [isLoggedIn, setLoggedIn] = useState(false);
+  const { isLoggedIn, authError } = useSelector((state) => state.handleLogin);
+  const dispatch = useDispatch();
   const [passwordValue, setPasswordValue] = useState("");
   const [emailValue, setEmailValue] = useState("");
 
   const userLogout = (e) => {
     e.preventDefault();
     auth.signOut().then(() => {
-      setIsAdmin(false);
-      setLoggedIn(false);
+      dispatch(logOut());
+      dispatch(popUpStatus(true));
     });
   };
 
   const userLogin = async (e) => {
     e.preventDefault();
-    const userLogin = await auth.signInWithEmailAndPassword(
-      emailValue,
-      passwordValue
-    );
-
-    const loggedInUserName = await db
-      .collection("users")
-      .doc(userLogin.user.uid)
-      .get();
-    setIsAdmin(loggedInUserName.data().isAdmin);
-    setLoggedIn(true);
+    try {
+      const userLogin = await auth.signInWithEmailAndPassword(
+        emailValue,
+        passwordValue
+      );
+      const loggedInUserName = await db
+        .collection("users")
+        .doc(userLogin.user.uid)
+        .get();
+      dispatch(
+        logIn(loggedInUserName.data().name, loggedInUserName.data().isAdmin)
+      );
+      dispatch(popUpStatus(true));
+    } catch (error) {
```

Comment:
you should wrap the smallest part possible in `try`, so only around the firebase auth line on 32, and the rest should error out naturally.